### PR TITLE
CAL-6 Added a NSILI Input Transformer for all NSIL views.

### PR DESCRIPTION
**Pull request by bdeining**
#### What does this PR do?
This PR adds an Input Transformer and Metacard Types for STANAG 4559 (NSILI) Federated Source responses.
#### Who is reviewing it?
@kcwire 
@jrnorth 
@dcruver 
@jaymcnallie 
@shane-voisard
@jlcsmith 
@beyelerb
#### How should this be tested?
This PR can be tested by building Alliance, installing the nsili-app in DDF, and federating to the mock nsili server.
#### Any background context you want to provide?
This PR is built on top of CAL-2 and CAL-1
#### What are the relevant tickets?
https://codice.atlassian.net/browse/CAL-6
https://codice.atlassian.net/browse/CAL-7
#### Checklist:
- [n/a ] Documentation Updated
- [x] Update / Add Unit Tests
- [n/a] Update / Add Integration Tests (integration tests will be added in a separate ticket)

### CODE REVIEW

File: catalog/nsili/catalog-nsili-source/src/main/java/org/codice/alliance/nsili/source/NsiliFilterDelegate.java [hunk 1]
```diff
@@ -0,0 +1,528 @@
...
+    public String propertyIsGreaterThan(String propertyName, double literal) {
+        String filter = filterFactory.buildPropertyIsGreaterThan(propertyName,
+                String.valueOf(literal));
+        return filter;
+    }
+
+    @Override
+    public String propertyIsGreaterThanOrEqualTo(String propertyName, String literal) {
+        String filter = filterFactory.buildPropertyIsGreaterThanOrEqual(propertyName,
+                EMPTY_STRING + SQ + literal + SQ);
```
Comment (jrnorth) [inline catalog/nsili/catalog-nsili-source/src/main/java/org/codice/alliance/nsili/source/NsiliFilterDelegate.java]: Don't need the `EMPTY_STRING` here.

File: catalog/nsili/catalog-nsili-source/src/main/java/org/codice/alliance/nsili/source/NsiliFilterDelegate.java [hunk 2]
```diff
@@ -0,0 +1,528 @@
...
+    public String propertyIsBetween(String propertyName, Date lowerBoundary, Date upperBoundary) {
+        String filter = filterFactory.buildPropertyIsBetween(propertyName, getStringFromDate(
+                lowerBoundary), getStringFromDate(upperBoundary));
+        return filter;
+    }
+
+    @Override
+    public String propertyIsBetween(String propertyName, int lowerBoundary, int upperBoundary) {
+        String filter = filterFactory.buildPropertyIsBetween(propertyName,
+                EMPTY_STRING + lowerBoundary,
```
Comment (jrnorth) [inline catalog/nsili/catalog-nsili-source/src/main/java/org/codice/alliance/nsili/source/NsiliFilterDelegate.java]: `String.valueOf()`?

File: catalog/nsili/catalog-nsili-source/src/main/java/org/codice/alliance/nsili/source/NsiliFilterDelegate.java [hunk 3]
```diff
@@ -0,0 +1,528 @@
...
+            filter = filterFactory.buildBeyondFilter(propertyName, wkt, distance);
+        }
+        return filter;
+    }
+
+    private String getStringFromDate(Date date) {
+        TimeZone timeZone = TimeZone.getTimeZone(UTC);
+        Calendar cal = Calendar.getInstance(timeZone);
+        cal.setTime(date);
+        SimpleDateFormat simpleDateFormat = new SimpleDateFormat(DATE_FORMAT);
```
Comment (jrnorth) [inline catalog/nsili/catalog-nsili-source/src/main/java/org/codice/alliance/nsili/source/NsiliFilterDelegate.java]: You might be interested in [`FastDateFormat`](http://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/time/FastDateFormat.html). It's thread-safe (unlike SimpleDateFormat), so you can create a static member instance and not have to create a new one each time this method is called.

File: catalog/nsili/catalog-nsili-source/src/main/java/org/codice/alliance/nsili/source/NsiliFilterDelegate.java (hunk 1 above)
Comment (bdeining) [inline catalog/nsili/catalog-nsili-source/src/main/java/org/codice/alliance/nsili/source/NsiliFilterDelegate.java]: Fixed in https://github.com/codice/alliance/pull/6.

File: catalog/nsili/catalog-nsili-source/src/main/java/org/codice/alliance/nsili/source/NsiliFilterDelegate.java (hunk 2 above)
Comment (bdeining) [inline catalog/nsili/catalog-nsili-source/src/main/java/org/codice/alliance/nsili/source/NsiliFilterDelegate.java]: Fixed in https://github.com/codice/alliance/pull/6.

File: catalog/nsili/catalog-nsili-source/src/main/java/org/codice/alliance/nsili/source/NsiliFilterDelegate.java (hunk 3 above)
Comment (bdeining) [inline catalog/nsili/catalog-nsili-source/src/main/java/org/codice/alliance/nsili/source/NsiliFilterDelegate.java]: Moved to FastDateFormat in https://github.com/codice/alliance/pull/6.

File: pom.xml [hunk 4]
```diff
@@ -18,7 +18,7 @@
     <artifactId>alliance</artifactId>
     <version>0.1-SNAPSHOT</version>
     <packaging>pom</packaging>
-    <name>DDF :: Alliance</name>
+    <name>Alliance</name>
     <description>Distributed Data Framework (DDF) Alliance</description>
```
Comment (jaymcnallie) [inline pom.xml]: Fix the description too. `Codice Alliance`?

File: catalog/nsili/pom.xml [hunk 5]
```diff
@@ -25,5 +25,8 @@
     <name>DDF :: Alliance :: NSILI</name>
```
Comment (jaymcnallie) [inline catalog/nsili/pom.xml]: Missed this one?

**Comment by jaymcnallie** [comment]
:+1:

**Comment by bdeining** [comment]
@jaymcnallie updated pom and description

**Comment by jaymcnallie** [comment]
Full build successful.  Merging..

**Pull request merged**